src/gui/transferlistwidget.cpp

```diff
@@ -487,7 +487,30 @@ void TransferListWidget::copySelectedNames() const
     qApp->clipboard()->setText(torrentNames.join('\n'));
 }
 
-void TransferListWidget::copySelectedHashes() const
+void TransferListWidget::copySelectedHashes(const CopyInfohashPolicy policy) const
+{
+    const auto selectedTorrents = getSelectedTorrents();
+    QStringList infoHashes;
+    infoHashes.reserve(selectedTorrents.size());
+    for (const BitTorrent::Torrent *torrent : selectedTorrents)
+    {
+        const auto infoHash1 = torrent->infoHash().v1();
+        const auto infoHash2 = torrent->infoHash().v2();
+        switch (policy)
+        {
+        case CopyInfohashPolicy::PreferVersion1:
+            infoHashes << (infoHash1.isValid() ? infoHash1.toString() : infoHash2.toString());
+            break;
+        case CopyInfohashPolicy::PreferVersion2:
+            infoHashes << (infoHash2.isValid() ? infoHash2.toString() : infoHash1.toString());
```

Comment:
As I've said before:

> Copying the ID, hashes, and magnet link from the context menu works fine. The v2 hash and torrent ID options should probably be greyed out for v1-only torrents though, as should the v1 hash option for v2-only.

And then the options should be renamed to just "info hash V1" and "info hash V2".

Reply:
Really I thought to care about cases when torrents of different versions (including hybrid) are mixed in selection.

Honestly I have no clear idea what exact purpose of such hash list.
Given the current state of the protocol, it seems that full sha-256 hashes are of no value to end users at all.

Reply:
> Given the current state of the protocol, it seems that full sha-256 hashes are of no value to end users at all.

I think they are: it's what appears in V2 magnet links and I would expect websites to list V2 hashes for V2 torrents.